cmake/thirdparty/get_spdlog.cmake

```diff
@@ -12,15 +12,39 @@
 # the License.
 # =============================================================================
 
-# Use CPM to find or clone speedlog
+# Use CPM to find or clone speedlog TODO: The logic here should be upstreamed into rapids-logger and
+# probably removed from rapids-cmake. Note that it is not possible to build with two different modes
+# for different projects in the same build environment, so this functionality must be kept
+# independent of the rapids_make_logger function. Alternatively, we could update the cpm calls for
+# spdlog to not promote targets to global scope and instead allow each project to find and configure
+# spdlog independently.
 function(find_and_configure_spdlog)
 
   include(${rapids-cmake-dir}/cpm/spdlog.cmake)
-  rapids_cpm_spdlog(
-    FMT_OPTION "EXTERNAL_FMT_HO"
-    INSTALL_EXPORT_SET rmm-exports
-    BUILD_EXPORT_SET rmm-exports)
-
+  if(RMM_SPDLOG_DYNAMIC)
+    rapids_cpm_spdlog(
+      FMT_OPTION "BUNDLED"
+      INSTALL_EXPORT_SET rmm-exports
+      BUILD_EXPORT_SET rmm-exports OPTIONS "SPDLOG_BUILD_SHARED ON" "BUILD_SHARED_LIBS ON")
+  else()
+    # For static spdlog usage assume that we want to hide as many symbols as possible, so do not use
+    # pre-built libraries. It's quick enough to build that there is no real benefit to supporting
+    # the alternative.
+    set(CPM_DOWNLOAD_spdlog ON)
+    rapids_cpm_spdlog(
+      # TODO: Is this safe to set up for all projects? Do we have to worry about the fmt patch
```

Comment:
Please check me on this. I've switched to the bundled fmt here so that we don't have to find fmt anymore.

Reply:
No longer relevant.